### DISCUSSION TITLE
Add lifecycle methods to ObservableRecipient

### DISCUSSION THREAD

**Discussion by alejandroortega08** [CommunityToolkit/Maui#2196] [New Feature Discussions]
It would be very useful if we included a powerful set of lifecycle methods. It would help track the state of the corresponding view and it can provide helper methods for more common functionality.

A good example would be a method called "Initialize" that is called after the native navigation has take place. It returns a Task, so can safely be marked async and used for heavy loading operations (such as hitting a backend API), without blocking the UI thread.

![Screenshot 2024-09-10 114004](https://github.com/user-attachments/assets/0cdad2f2-7ed4-45da-a5b4-33b05ad4e2ba)

https://www.mvvmcross.com/documentation/fundamentals/viewmodel-lifecycle

**Comment by bijington** [comment]
I think you want the repository over at https://github.com/communitytoolkit/dotnet as that is where `ObservableRecipient` lives
